Discord.Net.Abstractions/WebSocket/ISocketUserMessage.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Discord.WebSocket
+{
+    /// <inheritdoc cref="SocketUserMessage" />
+    public interface ISocketUserMessage : ISocketMessage, IUserMessage
+    {
+        /// <inheritdoc cref="SocketUserMessage.Reactions" />
+        new IReadOnlyDictionary<IEmote, IReactionMetadata> Reactions { get; }
+    }
+
+    /// <summary>
+    /// Provides an abstraction wrapper layer around a <see cref="WebSocket.SocketUserMessage"/>, through the <see cref="ISocketUserMessage"/> interface.
+    /// </summary>
+    public class SocketUserMessageAbstraction : SocketMessageAbstraction, ISocketUserMessage
+    {
+        /// <summary>
+        /// Constructs a new <see cref="SocketUserMessageAbstraction"/> around an existing <see cref="WebSocket.SocketUserMessage"/>.
+        /// </summary>
+        /// <param name="socketUserMessage">The value to use for <see cref="WebSocket.SocketUserMessage"/>.</param>
+        /// <exception cref="ArgumentNullException">Throws for <paramref name="socketUserMessage"/>.</exception>
+        public SocketUserMessageAbstraction(SocketUserMessage socketUserMessage)
+            : base(socketUserMessage) { }
+
+        /// <inheritdoc />
+        public IReadOnlyDictionary<IEmote, IReactionMetadata> Reactions
+            => SocketUserMessage.Reactions
+                .Select(x => (x.Key, Value: x.Value.Abstract()))
```

Comment:
Do you even need the `.Select()`? Couldn't you just move the logic into the `ToDictionary()` call?

Reply:
I probably could. Will try.